enterprise_access/apps/api/v1/views/assignment_configuration.py

```diff
@@ -240,14 +221,5 @@ def destroy(self, request, *args, uuid=None, **kwargs):
         # Custom delete() method should set the active flag to False.
         assignment_config_to_soft_delete.delete(reason=delete_reason)
 
-        # Manually unlink this soft-deleted AssignmentConfig from the associated policy.
```

Comment:
We stop doing this as the policy records are now soft deleted correct?

Reply:
Here's how I'm thinking about it:
1. An assignment-based policy record refers to an AssignmentConfiguration, call it config A.
2. For 👋 business reasons 👋 , we soft-delete config A.
3. The assignment-based policy type has a hard requirement of "my config can never be null".  So we leave the soft-deleted config A reference in the policy record.
4. We create config B, and then update the ref in the policy record from soft-deleted config A to new-and-fresh config B.

Reply:
From a domain-purity perspective, the assignment is not aware of the existence of policies, which is reason enough to remove this manual unlinking logic.  It's completely up to the policy itself to decide what to do when it discovers that its assignment config has been deleted. :+1: